### PR TITLE
Feature dynamic gbt

### PR DESCRIPTION
**Pull request by ramacarlucho**

### CODE REVIEW

File: include/bitprim/rpc/messages/mining/getblocktemplate.hpp [hunk 1]
```diff
@@ -168,7 +167,7 @@ nlohmann::json process_getblocktemplate(nlohmann::json const& json_in, Blockchai
 
     //TODO(fernando): check what to do with the 2018-May-15 Hard Fork
     //TODO(fernando): hardcoded 20000
-    if (getblocktemplate(result, error, error_code, libbitcoin::get_max_block_size() - 20000, std::chrono::seconds(30), chain)) {
+    if (getblocktemplate(result, error, error_code, libbitcoin::get_max_block_size() - 20000, std::chrono::seconds(5), chain)) {
```
Comment (dario-ramos) [inline include/bitprim/rpc/messages/mining/getblocktemplate.hpp]: Remove magic numbers

File: test/rpc.cpp [hunk 2]
```diff
@@ -213,12 +213,18 @@ class block_chain_dummy {
     //bool is_missing_previous_outputs(chain::transaction const& tx) const;
     ////    bool is_double_spent(chain::transaction const& tx) const;
 
-    ///// fetch_mempool_all()
-    using tx_mempool = std::tuple<libbitcoin::chain::transaction, uint64_t, uint64_t, std::string, size_t, bool>;
-
+#ifdef BITPRIM_CURRENCY_BCH
+    using tx_benefit = std::tuple<double /*benefit*/, size_t /*tx_sigops*/, size_t /*tx_size*/, size_t /*tx_fees*/, libbitcoin::data_chunk /*tx_hex*/, libbitcoin::hash_digest /*tx_id */> ;
```
Comment (dario-ramos) [inline test/rpc.cpp]: Use a class instead of a tuple

**Comment by mdallago** [comment]
Add logs for new blocks

File: include/bitprim/rpc/messages/mining/getblocktemplate.hpp (hunk 1 above)
Comment (mdallago) [reply include/bitprim/rpc/messages/mining/getblocktemplate.hpp]: pending

**Pull request closed without merging**